### PR TITLE
Align behavior for all Segmented Control modes

### LINKED ISSUE #2347: Segmented control - Align keyboard interactions between the different modes
The arrow key functionality for the default Segmented Control has been introduced by the Ionic 6 upgrade. Chip mode should use the same method as default mode (arrow keys once it has received focus).

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #2347

## What is the new behavior?
To implement similar keyboard interactions and behavior for all modes of the Segmented Control, it has been refactored to use `ion-segment` and `ion-segment-button` in all use cases.

This should also pave the way for removing the chip entirely, as it is no longer used internally, and was deprecated in #2341.  

🚨 Please note: The intented keyboard-interactions will only take effect when merged with the #2380. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.spec.ts
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.spec.ts:73]: This whole files diff makes the changes look more advanced than they actually are. 

What actually happens is that default mode is added to the testModes array, and then the whole `in 'default' mode` that looks deleted here is modified slightly to fit all modes. 

Then the other part of the test suite testing the chip mode is deleted.

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:31]: In general one of the only good use cases for using `unset` is when assigned to the `all` property. In all other cases I prefer to just assign the value that `unset` will result in. That would be either `initial` or `inherit` depending on the property.

In this case I would suggest assigning either `initial` or explicitly `none` (the initial value of `contain`)

```suggestion
      contain: initial; // Ensures the badge is not cut off
```

or 

```suggestion
      contain: none; // Ensures the badge is not cut off
```

https://developer.mozilla.org/en-US/docs/Web/CSS/unset
https://developer.mozilla.org/en-US/docs/Web/CSS/contain#formal_definition

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:29]: Just out of curiosity: Could the same be achieved by using `width: fit-content`?

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:47]: If you are declaring the same value for both axises anyway then:

```suggestion
      padding: #{utils.size('xxxs')};
```

**Review by jkaltoft — CHANGES_REQUESTED** [review]
Made a quick review of the style changes. Will follow up on the rest later 🙂

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.html
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.html:9]: 🙌🏻

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:31]: I just realised this may be code you just moved - not something you wrote, right?

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:29]:
```suggestion
      width: fit-content;
```

You should of course verify that my suggestion works as intended 😁

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:74]: If you're having issues with compact mode being too compact, you may be able to hit approximately the same amount of padding as before by changing these two custom properties. I think the closest you get is by using `15px` but since that is not one of our size tokens `size('s')` is probably an ok size.

If you choose `size('s')` you could even just remove both declarations, since it's the same as declared in [lines 19-20](https://github.com/kirbydesign/designsystem/blob/7e7c9e4da277169cd466b267501d6be102700f84/libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss#L19-L20). This of course requires that  `.sm` is used with compact mode.

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (jkaltoft) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:29]: > Just out of curiosity: Could the same be achieved by using `width: fit-content`?

https://github.com/kirbydesign/designsystem/pull/2417#discussion_r949005332

Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:29]: It indeed does, much more readable!

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:74]: Yeah, the reason for the compactness right now is actually that I wanted to propose a slightly _compacter_ mode, as there was very little difference before. But right now, I think it is just too compact.

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:47]: Great, didn't catch that when fiddling with the padding!

File: libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/segmented-control/segmented-control.component.scss:31]: You are right, just moved it, but lets change it anyways now that we have a good description of the context here! I have set it to `none`.

**Review by jkaltoft — APPROVED** [review]
Great work!

Let me know if you need a second review, e.g., if you make additional changes to the size of compact mode.

**Comment by RasmusKjeldgaard** [comment]
> Great work!
> 
> Let me know if you need a second review, e.g., if you make additional changes to the size of compact mode.

Only a very minor detail change, so merging this 🎉

**Pull request merged**
